src/Core/TimerTask/BasicTimerTask.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Chronos.Timer.Core
+{
+    /// <summary>
+    /// Represents a task that should be run at some point in the future.
+    /// </summary>
+    public class BasicTimerTask : ITimerTask
+    {
+        #region private 
+        private Action _task;
+        private bool _finished;
+        #endregion
+
+        #region properties
+        public TimeSpan Period { get; private set; }
+        public int TimesToExecute { get; private set; }
+
+        public bool CanExecute
+        {
+            get { return !_finished; }
+        }
+        public bool IsFinished
+        {
+            get { return _finished; }
+        }
+        #endregion
+
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="task">The job to run.</param>
+        /// <param name="period">How long until the first and subsequent executions.</param>
+        /// <param name="numTimesToRun">How long to repeat the task for (default is one execution).</param>
+        public BasicTimerTask(Action task, TimeSpan period, int numTimesToRun)
+        {
+            if(numTimesToRun == 0)
+            {
+                throw new ArgumentOutOfRangeException("numTimesToRun");
```

Comment:
Can you imbue the message with nameof(numTimesToRun) needs to be higher than 0.

Reply:
As per the doc, `ArgumentOutOfRangeException` is meant to only take in the param name. I'll put the info in the method documentation instead.

Reply:
You can use [this](https://docs.microsoft.com/en-us/dotnet/api/system.argumentoutofrangeexception.-ctor?view=netstandard-2.1#System_ArgumentOutOfRangeException__ctor_System_String_System_Object_System_String_) constructor for the exception :D